domain/application/state/charm.go

```diff
@@ -663,10 +666,11 @@ WHERE uuid = $charmID.uuid;
 		// If the charm requires sequencing, get the next revision from
 		// the reference name.
 		if requiresSequencing {
-			ch.Revision, err = s.sequence(ctx, tx, ch.ReferenceName)
+			rev, err := domainsequence.NextValue(ctx, s, tx, fmt.Sprintf("%s_%s", localCharmSquenceNamespace, ch.ReferenceName))
```

Comment:
Isn't this a breaking change? I'm basically asking about the case of a local charm that on 3.6 would have started with revision 1 and now it does with revision 0. 
I understand that it's not a dealbreaker for real world deployments, but what if there is some monitoring tool or anything that follows the revision? In that case this breaks it. I'm also thinking about integration tests that charmers use that might look at the rev.

Reply:
In 3.6 we start at 0